pipeline_dp/private_beam.py

```diff
@@ -104,3 +106,43 @@ def expand(self, pcol: pvalue.PCollection) -> pvalue.PCollection:
             value_extractor=self._sum_params.value_extractor)
 
         return dp_engine.aggregate(pcol, params, data_extractors)
+
+
+class Map(PrivatePTransform):
+    """Transform class for performing Map on PrivatePCollection."""
+
+    def __init__(self, fn: Callable, label: Optional[str] = None):
+        super().__init__(return_anonymized=False, label=label)
+        self._fn = fn
+
+    def expand(self, pcol: pvalue.PCollection):
+        return pcol | "map values" >> beam.Map(
+            lambda x: (self._privacy_id_extractor(x), self._fn(x)))
```

Comment:
It returns PCollection not PrivatePCollection. I'd suggest to do, similar way as it was done in [private_spark](https://github.com/OpenMined/PipelineDP/blob/main/pipeline_dp/private_spark.py).

Namely:

1.In `MakePrivate` to map `pcol` to PCollection `(privacy_id, value)` to pass such PCollection to the constructor of   `PrivatePCollection` .
2. In this class (i.e in Map) apply `fn` to values and to wrap the output `PCollection` by creating `PrivatePCollection` by calling the constructor of `PrivatePCollection`.
3. No need to pass `privacy_id_extractor` in `PrivatePCollection` anymore (elements of `PrivatePCollection` will be (privacy_id, value) so we will know privacy_id for each element).

WDYT?

Reply:
That's a good idea! Not having to pass the extractor all the way.